presto-cache/src/main/java/com/facebook/presto/cache/alluxio/AlluxioCachingFileSystem.java

```diff
@@ -69,7 +71,17 @@ public synchronized void initialize(URI uri, Configuration configuration)
             }
         });
         this.cacheQuotaEnabled = configuration.getBoolean(USER_CLIENT_CACHE_QUOTA_ENABLED.getName(), false);
-        localCacheFileSystem.initialize(uri, configuration);
+
+        // create URI with alluxio:// scheme for use with Alluxio
+        URI alluxioUri;
+        try {
+            alluxioUri = new URI(SCHEME, uri.getSchemeSpecificPart(), uri.getFragment());
+        }
+        catch (URISyntaxException e) {
+            throw new IOException("Unable to create Alluxio URI", e);
```

Comment:
Do you want to capture the URI that failed as part of the exception ?

something like "Unable to create Uri for {uri}"

Reply:
This is a core change, we should test well and make sure our internal use-cases are working with caching. I would recommend testing first before we merge this PR.